In [1]:
import numpy as np 
import pandas as pd 
from scipy import stats


## Q1

In [3]:
from google.colab import files
uploaded=files.upload()

Saving Cutlets.csv to Cutlets.csv


In [4]:
ctl=pd.read_csv('Cutlets.csv')

In [6]:
ctl.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [7]:
ctl.count()

Unit A    35
Unit B    35
dtype: int64

In [11]:
UnitA=pd.Series(ctl.iloc[:,0])
UnitA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [14]:
UnitB=pd.Series(ctl.iloc[:,1])
UnitB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [17]:
stats.t.cdf(UnitA,UnitB)

array([0.99985434, 0.99986819, 0.99986373, 0.99990568, 0.99993561,
       0.99991363, 0.99989735, 0.99987397, 0.99987862, 0.99988   ,
       0.99991824, 0.99994913, 0.99986792, 0.99978169, 0.99985915,
       0.99987971, 0.99981214, 0.99993765, 0.999933  , 0.99990678,
       0.99988036, 0.99993034, 0.99978209, 0.9999266 , 0.99987527,
       0.99987499, 0.99991331, 0.99985129, 0.99982554, 0.99993131,
       0.99987884, 0.99992354, 0.99988643, 0.99989838, 0.99991301])

In [15]:
#2-sample 2-tail t-test
p_value=stats.ttest_ind(UnitA,UnitB)
p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

## hence p>0.05
## inference : Accept Null hypothesis
## H0: u1, u2 there is no difference in diameter of cutlets between two units. 

## Q2

In [18]:
from google.colab import files 
uploaded= files.upload()

Saving LabTAT.csv to LabTAT.csv


In [19]:
lab=pd.read_csv('LabTAT.csv')

In [20]:
lab.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [21]:
#Anova f-test stastistics of oneway
p_value=stats.f_oneway(lab.iloc[:,0],lab.iloc[:,1],lab.iloc[:,2],lab.iloc[:,3])
p_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

## p = 2.1156708949992414e-57
## alpha = 0.05
## comparing p and alpha
So we get
## p< alpha
Inference: Reject Null Hypothesis Ho.
Thus, Accept Alternate Hypothesis HA as It has Variance: Atleast one sample TAT population mean is different.

## Q3

In [22]:
from google.colab import files
uploaded=files.upload()

Saving BuyerRatio.csv to BuyerRatio.csv


In [23]:
sale=pd.read_csv('BuyerRatio.csv')

In [25]:
sale.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [26]:
sale_table=sale.iloc[:,1:6]
sale_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [27]:
sale_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [30]:
val=stats.chi2_contingency(sale_table)
val

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [31]:
type(val)

tuple

In [35]:
no_of_rows=len(sale_table.iloc[0:2,0])
no_of_columns=len(sale_table.iloc[0,0:4])
degree_of_f=(no_of_rows-1)*(no_of_columns-1)
print('DOF',degree_of_f)

DOF 3


In [36]:
Expected_value=val[3]
Expected_value

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

In [38]:
chi_square=sum([(o-e)**2/e for o, e in zip(sale_table.values,Expected_value)])
chi_square_stastestic=chi_square[0]+chi_square[1]
chi_square_stastestic

1.5152956451130446

In [44]:
from scipy.stats.distributions import chi2
critical_value=chi2.ppf(0.95,3)
critical_value=chi2.ppf(0.95,3)
critical_value

7.814727903251179

In [43]:
p_value=1-chi2.cdf(chi_square_stastestic,3)
p_value

0.6787446296467897

## P=0.6787446296467897
## Alpha=0.05
## comparing p and alpha
So we get
## p> alpha
Inference: Accept Null Hypothesis H0
So, all proportions are equal.

In [45]:
from google.colab import files
uploaded=files.upload()

Saving Costomer+OrderForm.csv to Costomer+OrderForm.csv


In [46]:
form=pd.read_csv('Costomer+OrderForm.csv')

In [47]:
form.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [48]:
form.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [49]:
form.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [50]:
form.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [51]:
form.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [52]:
dp=np.array([[271,267,269,280],[29,33,31,20]])
dp

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [53]:
stats.chi2_contingency(dp)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

## p = 0.2771020991233135
## alpha = 0.05
## comparing p and alpha
So we get
## p> alpha
Inference: Accept Null Hypthesis HO i.e. Independence of categorical variables.
Thus, customer order forms defective % does not varies by centre.